src/index.ts

```diff
@@ -1,9 +1,3 @@
-/**
- * Example function that returns a greeting for the given name.
- *
- * @param name - The name to greet.
- * @returns The greeting.
- */
-export default function greeter(name: string): string {
-  return `Hello, ${name}!`;
-}
+export * from './provider-from-engine';
+export * from './provider-from-middleware';
+export type { SafeEventEmitterProvider } from './safe-event-emitter-provider';
```

Comment:
I'm unsure whether we need to export the class itself. I opted just for exporting the type for now, to match what we do today in `eth-json-rpc-middleware`.

Reply:
Seems reasonable since we only instantiate it here ATM